### DISCUSSION TITLE
community for fontforge

### DISCUSSION THREAD

**Discussion by brigazvi** [fontforge/fontforge#5691] [Show and tell]
I don't know if its the right place. but, i'll give it a try.
I'm not a programmer. so i fill a little bit rude to come here to all the hard-workers and tell them all my thoughts about ways to improve fontforge. So before all thing, I would like to say how much I appreciate all your efforts to make fontforge better.

Now, I think that one of the biggest problems we face in fontforge development is the lack of enough developers and enough time to develop.

I think there is three ways to try solving it:
1. making it easy for news contributors to understand the code and how they can contribute.
2. growing the community around fontforge. new users means (sometimes) new developers.
3. fund. like https://fund.blender.org/. making it easy to current developers to spend more time on the project.

I don't have a clue about programming and how to make it easy to newcomers.
about fund i don't know either. (I know there it Patreon but they take a share. maybe there is better platform)
but maybe I can help with the community. maybe its a good start to open a community repository here in github for ideas. we can make posts in the website about designers who use fontforge. etc.

what do you think, guys?

(and sorry for my english)

**Comment by skef** [comment]
The primary reason FontForge doesn't have many active developers is because the code is difficult to work with. It's written in an old style of C with little abstraction, and uses a custom UI widget set. Education can help with that but only a little. Funding can help but past efforts haven't funded anything like typical market rates for substantial work.

One hope is that new contributions can be coded in python rather than C. There are many things that are not possible to do via the Python API but many more that are possible. Unfortunately there is much less code sharing of python extensions than one would hope.

The next release should include some new features to help with that, specifically #4642, #4671, and #4682.  (Small API improvements like #4646 help as well.) The plugin API will make it much easier to share extensions -- a user can just install the relevant python package and then activate the plugin. The menu improvements will make having a lot of extensions in the Tools menu a bit less confusing. And multiple question dialogs can be used as a basic API to configure an extension.

I'm just one developer but my advice would be to wait for this next release and then try to help us publicize these improvements to kick-start a larger python FontForge development community.
